### PR TITLE
Overhaul of MODData

### PR DESCRIPTION
**Pull request by ml-evs**
This WIP PR overhauls the MODData class with the following aims:

- [ ] easier for the user to select featurizers (probably now pushed to the next PR)
- [x] make RR and NMI calculations more modular so they can be tested more effectively
- [x] allow for easier re-use of old calculations, and checkpointing of new descriptor calculations
- [x] some general tidying
- [x] Added integration test that featurisation returns same features as older versions used for the paper

### CODE REVIEW

File: modnet/preprocessing.py
Comment (davidwaroquiers) [inline modnet/preprocessing.py:800]: Could now use get_features_relevance_redundancy here. Maybe some checks to be performed.

**Comment by davidwaroquiers** [comment]
Good work. Good to see things are going on :+1:

File: modnet/preprocessing.py
Comment (ml-evs) [reply modnet/preprocessing.py:800]: Yep, my plan is to add tests for the current API then switch things over adiabatically :)

**Comment by davidwaroquiers** [comment]
Just a comment out of a few discussions with @ppdebreuck and Gian-Marco, it might be good to allow for other forms of functions than the one that is currently implemented, i.e. not restricting ourselves to nmi/(maxcrossnmi^p+c) but being able to directly pass a callable.

**Comment by ml-evs** [comment]
Apologies for the delay, my tests now exactly match the current output of MODData (down to the values of features, and which features remain in `df_featurized` after cleaning). I had to tweak some more field names that had previously been hardcoded to other values, but we can soon move all of this to the "presets" package so that the paper can be reproduced as its stands.

**Comment by ml-evs** [comment]
Once we're ready to merge this I'll also squash this down to fewer commits so I'm not flooding the commit log, perhaps @ppdebreuck you could just confirm the changes in the last couple of commits as they are?

File: modnet/preprocessing.py
Comment (ml-evs) [reply modnet/preprocessing.py:821]: Just to answer this more, the calculation of NMI needs to be reworked before we can test it easily, I think following the discussion yesterday we should try to merge this now so its not holding up other work.

**Comment by ml-evs** [comment]
Looks like there's just one more test failure on the CI to fix (passes locally)

**Comment by ml-evs** [comment]
So the problem is caused by the following: unfortunately the latest pymatgen/scipy versions fill null values in partial RDFs slightly differently, so this feature gets dropped every time. For now, I've pinned the version of pymatgen to maintain consistency with the paper.

**Review by ppdebreuck — APPROVED** [review]
Reviewed changes, great! => Merging PR

**Pull request merged**
